packages/ts-client/src/webrtc/tracks/LocalTrack.ts

```diff
@@ -149,25 +151,27 @@ export class LocalTrack<EndpointMetadata, TrackMetadata> implements TrackCommon
       stream?.addTrack(newTrack);
     }
 
-    if (this.trackContext.track && !newTrack) {
-      const mediaEvent = generateMediaEvent('muteTrack', { trackId: trackId });
-      webrtc.sendMediaEvent(mediaEvent);
-      webrtc.emit('localTrackMuted', { trackId: trackId });
-    } else if (!this.trackContext.track && newTrack) {
-      const mediaEvent = generateMediaEvent('unmuteTrack', {
-        trackId: trackId,
-      });
-      webrtc.sendMediaEvent(mediaEvent);
-      webrtc.emit('localTrackUnmuted', { trackId: trackId });
-    }
-
+    this.trackContext.track = newTrack;
     this.mediaStreamTrackId = newTrack?.id ?? null;
 
+    if (action === 'mute') {
+      emitEvents('mute', webrtc, trackId);
+    } else if (action === 'unmute') {
+      emitEvents('unmute', webrtc, trackId);
+    }
+
     try {
       await this.sender.replaceTrack(newTrack);
-      this.trackContext.track = newTrack;
     } catch (_error) {
-      // ignore
+      // rollback: emit opposite events and revert internal state
+      if (action === 'mute') {
+        emitEvents('unmute', webrtc, trackId);
+      } else if (action === 'unmute') {
+        emitEvents('mute', webrtc, trackId);
+      }
```

Comment:
**praise:** 
```suggestion
      const rollbackAction = (action === 'mute' && 'unmute' ) || (action === 'unmute' && 'mute' );
      if (rollbackAction) emitEvents(rollbackAction, webrtc, trackId);
```

Reply:
It's nice, because it's shorter, but I'll stick to my simple approach 😅